### PR TITLE
Moving WCF Client/CoreWCF packages to extension-wcf folder to fix pipeline failure in Docx stage

### PR DESCRIPTION
**Pull request by subhra231**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by jsquire** [comment]
Please also update the comment [here](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L364) to reflect the new location.

**Comment by jsquire** [comment]
**note:** I've cleaned up the `extensions - wcf` pipelines from the previous directory.

**Comment by jsquire** [comment]
New pipelines have been created, classified, and enabled.

**Comment by subhra231** [comment]
> Please also update the comment [here](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props#L364) to reflect the new location.

Updated this location.

File: sdk/extension-wcf/ci.yml
Comment (jsquire) [reply sdk/extension-wcf/ci.yml:30]: You have two packages in this subdirectory.   You have one artifact entry here.   That looks to be one of your CI failures.   Aren't we publishing `Microsoft.WCF.Azure.StorageQueues`?

Comment (mconnew) [reply sdk/extension-wcf/ci.yml:30]: @jsquire, I'm seeing the following in this PR, is there something else beyond this that's needed?

```yml
    Artifacts:
    - name: Microsoft.CoreWCF.Azure.StorageQueues
      safeName: MicrosoftCoreWCFAzureStorageQueues
    - name: Microsoft.WCF.Azure.StorageQueues
      safeName: MicrosoftWCFAzureStorageQueues
```

**Comment by jsquire** [comment]
LGTM.   I kicked off the CI and Live pipelines.  Once we've confirmed they're passing, we should be good to merge.

**Comment by jsquire** [comment]
Please add a CODEOWNERS section for this new path as well.   I'll create the "Extensions - WCF" label for it.

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:968]: Please move this up to with the other non-management services, preserving alphabetical order by tag name.

File: .github/CODEOWNERS
Comment (jsquire) [inline .github/CODEOWNERS:969]:
```suggestion
# PRLabel: %Extension - WCF
/sdk/extension-wcf/                                                @mconnew @subhra231

# ServiceLabel: %Extension - WCF
# ServiceOwners:                                                   @mconnew @subhra231
```

Comment (jsquire) [reply .github/CODEOWNERS:969]: You'll need an entry for PR and support use.   This format should work.

File: .github/CODEOWNERS
Comment (subhra231) [reply .github/CODEOWNERS]: I have moved this to # ######## Services ########. Please let me know if this is still not the right place.

**Comment by jsquire** [comment]
@subhra231:  Changes look good to me.   I've kicked off the two pipelines.  Once those are passing, I'll approve and merge.

**Comment by jsquire** [comment]
@subhra231:  Looks like you've got some environmental issues with Azurite that you'll need to work through.   Please let me know when you're ready and I'll kick the pipelines again.

**Comment by subhra231** [comment]
> @subhra231: Looks like you've got some environmental issues with Azurite that you'll need to work through. Please let me know when you're ready and I'll kick the pipelines again.

@jsquire Thats right. We noticed the issue and we are taking a look. We will let you know when we have a fix. Thanks

**Comment by subhra231** [comment]
/azp run prepare-pipelines

> > @subhra231: Looks like you've got some environmental issues with Azurite that you'll need to work through. Please let me know when you're ready and I'll kick the pipelines again.
> 
> @jsquire Thats right. We noticed the issue and we are taking a look. We will let you know when we have a fix. Thanks

@jsquire Looks like the azurite test template was missing from tests.yml file. Submitted the change. hopefully that fixes this failure

File: sdk/extension-wcf/ci.yml
Comment (weshaggard) [inline sdk/extension-wcf/ci.yml:36]: Do you still need these steps for the ci pipeline?

Comment (subhra231) [reply sdk/extension-wcf/ci.yml:36]: Yes we need that step, i have reverted these changes.

**Comment by jsquire** [comment]
Looks like whatever is going on with the macOS hangs is consistent and repros for each run.   From the pipeline output, my best guess is that a test is hanging and doesn't have a forced timeout.   We'll need to figure out which test that is and either fix or ignore it for now.   That looks like the last blocker for getting this merged.

Going forward, I'd suggest implementing a timeout in all of the tests that have background assumptions and which aren't governed by a network operation timeout.    Event Hubs is a decent [example]()https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.MainProcessingLoop.cs#L42) of one approach for doing so.

**Comment by mconnew** [comment]
@jsquire, the issue turned out to be that we were using the default config for aspnetcore which listens for https requests too. It was trying to open the users certificate store when running on MacOS which popped up the MacOS equivalent of a UAC window asking for the user password to unlock the certificate store. Oddly it only did it in some configs. The fix was to not use the default host builder and create it manually.

**Comment by jsquire** [comment]
> @jsquire, the issue turned out to be that we were using the default config for aspnetcore which listens for https requests too. It was trying to open the users certificate store when running on MacOS which popped up the MacOS equivalent of a UAC window asking for the user password to unlock the certificate store. Oddly it only did it in some configs. The fix was to not use the default host builder and create it manually.

Cool.  I'll kick the pipelines off again and if all is well, we'll get this merged in.

**Review by jsquire — APPROVED** [review]

**Pull request merged**
